fabfile/compat.py

```diff
@@ -0,0 +1,31 @@
+from fabric import Connection
+from invoke import Context
+from invoke.exceptions import Exit
+
+from connection import all_connections
+
+def is_coordinator_connection(c):
+    # same as @roles('master') in fabric 1.14
+
+    if isinstance(c, Connection):
+        return c.host == 'localhost'
+    elif isinstance(c, Context):
+        return True
+    raise Exit('Unexpected connection!')
+
+def execute_on_all_nodes_if_no_hosts(c, callback, *args, **kwargs):
+    # we want to to behave same with fabric 1.14 fab <task> will execute on all nodes
+    # but in current version it will only execute on localhost without connection using context
+    # IMPORTANT: for each @task, we should run that method on the first line
```

Comment:
This is the trickiest work in the PR. 

Previously when we `fab <taskA>`, it will run on all nodes.
After fabric upgrade, when we `fab <taskA>`, it will run only on local node.

But it would be a pain to specify -H host1, host2, .... to run on all nodes. Thus, I mimicked the old behaviour. If we can find a better way to run on all nodes if -H is not specified, it would be great. Otherwise, we need that tricky stuff on the first line of each task to intercept connection object to see if it is called from `fab <taskA>`.